src/items/associated-items.md

```diff
@@ -93,9 +93,34 @@ Associated functions whose first parameter is named `self` are called *methods*
 and may be invoked using the [method call operator], for example, `x.foo()`, as
 well as the usual function call notation.
 
-If the type of the `self` parameter is specified, it is limited to the type
-being implemented (or `Self`), or a reference or mutable reference to the
-type, or a boxed value of the type being implemented (such as `Box<Self>`).
+If the type of the `self` parameter is specified, it is limited to one of the
+following types:
+
+- `Self`
+- `&Self`
+- `&mut Self`
+- [`Box<Self>`]
+- [`Rc<Self>`]
+- [`Arc<Self>`]
+
+The `Self` portion of the type may be replaced with the type being
+implemented.
```

Comment:
Hmm, this feels a bit of an awkward way to say "the way type of the receiver is written doesn't matter", but I'm not sure that's any better.